CRM/ACL/API.php

```diff
@@ -113,18 +113,18 @@ public static function whereClause(
       }
     }
 
-    if (!$contactID) {
-      $contactID = CRM_Core_Session::getLoggedInContactID();
-    }
-    $contactID = (int) $contactID;
-
     // first see if the contact has edit / view all permission
     if (CRM_Core_Permission::check('edit all contacts', $contactID) ||
       ($type == self::VIEW && CRM_Core_Permission::check('view all contacts', $contactID))
     ) {
       return $deleteClause;
     }
 
+    if (is_null($contactID)) {
```

Comment:
So this change basically permits 0 to be passed in - why is it moved below the other?

Reply:
@eileenmcnaughton the problem was that this function wasn't differentiating `NULL` from `0` and they are very different.

`0` = no user
`null` = current user

I fixed that and then moved the block down because `CRM_Core_Permission::check` already handles `NULL` and `0` correctly so there was no need to mess with the `$contactID` value before calling that function.